viamonvif/service.go

```diff
@@ -287,6 +312,53 @@ func downloadPreviewImage(ctx context.Context, logger logging.Logger, snapshotUR
 	return dataURL, nil
 }
 
+// fetchImageFromRTSPURL fetches the image from the rtsp URL and returns it as a data URL.
+func fetchImageFromRTSPURL(ctx context.Context, logger logging.Logger, rtspURL string) (string, error) {
+	// Wrap viamrtsp.Config in a resource.Config
+	rtspConfig := viamrtsp.Config{
+		Address: rtspURL,
+	}
+	resourceConfig := resource.Config{
+		Name:                "camera_name", // Replace with the actual camera name
+		API:                 camera.API,
+		Model:               viamrtsp.ModelAgnostic,
+		ConvertedAttributes: &rtspConfig,
+	}
+
+	camera, err := viamrtsp.NewRTSPCamera(ctx, nil, resourceConfig, logger)
+	if err != nil {
+		return "", fmt.Errorf("failed to create RTSP camera: %w", err)
+	}
+	defer func() {
+		if closeErr := camera.Close(ctx); closeErr != nil {
+			logger.Warnf("failed to close camera: %v", closeErr)
+		}
+	}()
+
+	retryInterval := rtspImageInterval
+	timeout := rtspPollTimeout
+	ticker := time.NewTicker(retryInterval)
+	defer ticker.Stop()
+	timeoutChan := time.After(timeout)
+	var imageErr error
+	for {
+		select {
```

Comment:
is this `select` missing a `ctx.Done()` case?

Reply:
yes it is :)

Reply:
Yes thanks, added `ctx.Done` case